tests/src/core/testqgsogcutils.cpp

```diff
@@ -190,25 +190,25 @@ void TestQgsOgcUtils::testExpressionFromOgcFilterWFS20_data()
 
   QTest::newRow( "bbox coordinates" ) << QString(
     "<Filter>"
-    "<BBOX><ValueReference>Name>NAME</ValueReference><gml:Box srsName='foo'>"
+    "<BBOX><ValueReference>Name>NAME</ValueReference><gml:Box xmlns:gml=\"http://www.opengis.net/gml\" srsName=\"foo\">"
```

Comment:
I have similar concerns as https://github.com/qgis/QGIS/pull/61878#discussion_r2094201746 -- let's finalise the discussion there though

Reply:
ok, I've reworked the pull request, taking the strategy done in QgsOgcUtils::geometryFromGML() that is to wrap the XML content in a node that declares missing namespaces.
- In the QGIS code, this is done for the geom_from_gml() and ST_GeomFromGML()  filter function. OK here
- In the test, this is done before calling doc.setContent(), so that XML parsing done in the test code itself doesn't fail (this is slightly less invasive than doing the QT version dependent injection of namespaces)
- In the test, change the expected output of some tests where namespaces are now output, which should be fine